src/core/retargeting_engine/python/interface/base_retargeter.py

```
@@ -281,71 +371,61 @@ def _validate_inputs(self, inputs: RetargeterIO) -> None:
             return cached
 
-        # Get inputs from context (if any are needed)
-        if len(self._inputs) > 0:  # Only look for inputs if this module has any
-            inputs = context.get_leaf_input(self.name)
+        if len(self._inputs) > 0:
+            inputs = cache.get_leaf_input(self.name)
             if inputs is None:
-                raise ValueError(f"Input '{self.name}' not found in context")
-
-            inputs = self._fill_optional_inputs(inputs)
+                raise ValueError(f"Input '{self.name}' not found in cache")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)